src/providers/virtual/qgsvirtuallayersourceselect.cpp

```diff
@@ -229,13 +234,37 @@ void QgsVirtualLayerSourceSelect::testQuery()
         QMessageBox::warning( nullptr, tr( "Test Virtual Layer " ), tr( "The unique identifier field <b>%1</b> was not found in list of fields:<ul>%2</ul>" ).arg( mUIDField->text(), bulletedFieldNames.join( ' ' ) ) );
       }
       else
-        QMessageBox::information( nullptr, tr( "Test Virtual Layer" ), tr( "No error" ) );
+      {
+        if ( mGeometryRadio->isChecked() && mCRS->text().isEmpty() )
```

Comment:
This is a rather small function so please feel free to ignore the suggestion but I find it more readable when a function has only one entry and one exit point, as suggested below:
```suggestion
        bool result = false;
        ...
        result = (!(mGeometryRadio->isChecked() && mCRS->text().isEmpty()) ||
                  (QMessageBox::Yes == QMessageBox::question(...)))
        ...
        return result;
```

Reply:
@asimrp thanks for you review. I'm not a very experienced c++ coder, so I came up with this:

```
        result = ( !( mGeometryRadio->isChecked() && mCRS->text().isEmpty() ) ||
                   ( QMessageBox::Yes == QMessageBox::question( nullptr, tr( "Test Virtual Layer " ), tr( "No CRS defined, are you sure you want to create a layer without a crs?" ), QMessageBox::Yes | QMessageBox::No ) ) );
```
But although concise, I personally(!) cannot read this enough to see if the outcome is the same as the (indeed maybe too long if statements...
Not sure about real c++ coders, but I think I prefer to leave it like this (but others feel free to create cleaner code).